pkg/apis/edge/v1alpha1/edge-placement.go

```diff
@@ -20,14 +20,23 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EdgePlacement binds a collection of (a) Namespaces and non-namespaced objects
-// to (b) a collection of Locations.
-// An EdgePlacement object appears in the center and directs all the syncable
-// objects in the selected Namespaces and all the selected non-namespaced objects
-// to propagate to _all_ of the selected Locations
-// (one SyncTarget in each such Location).
-// This is not entirely unrelated to a TMC Placement, which directs the selected
-// Namespaces to propagate to _one_ of the selected Locations.
+// EdgePlacement exists in the center and binds (a) a collection of
+// Locations with (b) both (b1) objects in the center to downsync
+// (propagate desired state from center to edge and return reported
+// state from edge toward center), and (b2) a way of identifying objects
+// (in edge clusters) to upsync (propagate from edge toward center).
+// Both downsyncing and upsyncing are with all of the Locations.  This
+// is not entirely unrelated to a TMC Placement, which directs the
+// selected Namespaces to propagate to _one_ of the selected
+// Locations.
+//
+// The objects to downsync are those in selected namespaces plus
```

Comment:
@MikeSpreitzer non-namespace objects can be specified by group, kind, name, and label. why namespace objects is not specified like that? should we assume all the non-namespace objects under specified namespaces are downsynced?

Reply:
For example, we have use case of downsyncing a secret from mailbox to p-cluster by edge syncer, and want to specifiy specific secret to be downsynced, but not all secrets on the mailbox.

Reply:
Can I please start by asking for a clarification of what you mean by "non-namespace"?  I can think of two meanings: (1) not a namespace, and (2) not _in_ a namespace.  The latter I sometimes call "non-namespaced", and it is also called "cluster-scoped" in the general Kubernetes community.

I based my proposal for EdgePlacement on TMC's Placement.  There (https://github.com/kcp-dev/kcp/blob/v0.11.0/pkg/apis/scheduling/v1alpha1/types_placement.go#L78) you will see that there is only a label selector for `Namespace` objects, and everything in each matching namespace gets downsynced.

Does something go wrong for C2P with the current downsync semantics for EdgePlacement and the proposal here for adding upsync?

I should perhaps expand on my answer to @pdettori 's earlier question (https://github.com/kcp-dev/edge-mc/pull/239#discussion_r1139596912).  I am trying to not make the upsync spec more complicated than necessary because, as observed by him in other conversations, we have to come up with implementations (possibly one connecting to TMC, certainly one in EMC).  I do not want to make our initial implementation job harder than it needs to be.  Nothing prevents adding complexity later if and when a need arises.

Reply:
OK, let's revisit this later, since this might be changed if we have new edge syncer.

Reply:
@MikeSpreitzer I am ok with keeping the specs simpler for now.